pkg/controller/deployment/deployment_controller.go

```diff
@@ -1294,3 +1413,27 @@ func (dc *DeploymentController) rollbackToTemplate(deployment *extensions.Deploy
 	d, err = dc.updateDeploymentAndClearRollbackTo(deployment)
 	return
 }
+
+// isScalingEvent checks whether the provided deployment has been updated with a scaling event
+// by looking at the desired-replicas annotation in the active replica sets of the deployment.
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
+	if err != nil {
+		return false
+	}
+	allRSs := append(oldRSs, newRS)
+	for _, rs := range controller.FilterActiveReplicaSets(allRSs) {
+		if desiredString, ok := rs.Annotations[deploymentutil.DesiredReplicasAnnotation]; ok {
```

Comment:
A helper is fine.  Will probably also want a helper such as UpdateRsAnnotationForDeployment(rs, dep) since it is pretty common to copy the Replica count from dep onto the rs annotations